fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/MetricLoader.java

```diff
@@ -18,7 +21,12 @@
      * @deprecated in favor of loadMetricDictionary(MetricDictionary, DimensionDictionary)
      */
     @Deprecated
-    void loadMetricDictionary(MetricDictionary metricDictionary);
+    default void loadMetricDictionary(MetricDictionary metricDictionary) {
+        String message = "The method loadMetricDictionary(MetricDictionary, DimensionDictionary) must be implemented " +
+                "for MetricLoader";
+        LoggerFactory.getLogger(DimensionToNameSerializer.class).error(message);
```

Comment:
'must' is too strong here as well.  There are many perfectly healthy MetricLoaders that don't implement the new method, and I don't really think we should be strong arming them to do so.  Also, this should not throw a generic runtime exception.
https://docs.oracle.com/javase/7/docs/api/java/lang/UnsupportedOperationException.html